src/adapters/source-adapters/OSMSourceAdapter.sol

```diff
@@ -1,19 +1,22 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity 0.8.17;
 
-import {SnapshotSource} from "./SnapshotSource.sol";
+import {DiamondRootOval} from "../../DiamondRootOval.sol";
+import {SnapshotSourceLib} from "../lib/SnapshotSourceLib.sol";
 import {IOSM} from "../../interfaces/makerdao/IOSM.sol";
 
 /**
  * @title OSMSourceAdapter contract to read data from MakerDAO OSM and standardize it for Oval.
  */
-abstract contract OSMSourceAdapter is SnapshotSource {
+abstract contract OSMSourceAdapter is DiamondRootOval {
     IOSM public immutable osmSource;
 
     // MakerDAO performs decimal conversion in collateral adapter contracts, so all oracle prices are expected to have
     // 18 decimals and we can skip decimal conversion.
     uint8 public constant decimals = 18;
 
+    SnapshotSourceLib.Snapshot[] public osmSnapshots; // Historical answer and timestamp snapshots.
```

Comment:
also this works well I think. when you need the snapshot within the adapter you simple: a) import the lib b) have a local snapshots array.